app/controllers/shopify_app/sessions_controller.rb

```diff
@@ -94,17 +94,20 @@ def authenticate_with_partitioning
 
     # rubocop:disable Lint/SuppressedException
     def set_user_tokens_option
-      if shop_session.blank?
+      store_session = shop_session ||
```

Comment:
Is this safe? A nefarious merchant could put someone else's shop name in the query params and then we would make some API call against that other shop?

Reply:
You're right. Since this check would happen on the top level, we should allow the use of the cookie in this case.